project/src/main/java/com/google/sps/servlets/EventCreationServlet.java

```diff
@@ -96,11 +96,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Event event = new Event.Builder(name, description, labels, location, date, time, host).build();
     if (eventId == null) {
       SpannerTasks.insertorUpdateEvent(event);
-      searchStore.addEventToIndex(event.getId(), name, description);
     } else {
       event = event.toBuilder().setId(eventId).build();
       SpannerTasks.insertorUpdateEvent(event);
+      SpannerTasks.deleteIndexEntriesByEventId(eventId);
```

Comment:
Can we just update index instead?

Reply:
I tried to think of a way but in the results table we have <EventID, KeywordID, Ranking> and we have to run the algorithm to get the new keywords. If the new keywords are of a different number of rows than the old one, no way to update without deleting or inserting depending on the difference.